### PR TITLE
Weed out dependencies on librustc_llvm and librustc.

### PR DESCRIPTION
**Pull request by eddyb**
Found while working on #32570. cc @nikomatsakis

### CODE REVIEW

**Comment by eddyb** [comment]
@oli-obk Was there a specific reason `rustc_const_eval` ended up in `TARGET_CRATES`? Don't want to break anything with this change.

**Comment by oli-obk** [comment]
I was sure it was needed at some point, but I don't remember why.
Probably I misunderstood something about how staging and
plugin-unit-tests work.

I don't see why it should be there.

Am 29.03.2016 um 12:56 schrieb Eduard-Mihai Burtescu:

**Comment by arielb1** [comment]
This breaks the `execution-engine` test:

```
test.rs:31:5: 31:16 error: unresolved import `rustc::llvm`. There is no `llvm` in `rustc` [E0432]
test.rs:31 use rustc::llvm;
               ^~~~~~~~~~~
test.rs:31:5: 31:16 help: run `rustc --explain E0432` to see a detailed explanation
test.rs:105:12: 105:24 error: the type of this value must be known in this context
test.rs:105         if ee.is_null() {
                       ^~~~~~~~~~~~
test.rs:100:21: 100:25 error: the trait `core::marker::Sized` is not implemented for the type `[std::path::PathBuf]` [E0277]
test.rs:100         let (llmod, deps) = compile_program(program, sysroot.clone())
                                ^~~~
test.rs:100:21: 100:25 help: run `rustc --explain E0277` to see a detailed explanation
test.rs:100:21: 100:25 note: `[std::path::PathBuf]` does not have a constant size known at compile-time
test.rs:100:21: 100:25 note: all local variables must have a statically known size
test.rs:120:21: 120:25 error: the trait `core::marker::Sized` is not implemented for the type `[std::path::PathBuf]` [E0277]
test.rs:120         let (llmod, deps) = compile_program(program, self.sysroot.clone())
                                ^~~~
test.rs:120:21: 120:25 help: run `rustc --explain E0277` to see a detailed explanation
test.rs:120:21: 120:25 note: `[std::path::PathBuf]` does not have a constant size known at compile-time
test.rs:120:21: 120:25 note: all local variables must have a statically known size
test.rs:136:17: 136:29 error: the type of this value must be known in this context
test.rs:136             if !fv.is_null() {
                            ^~~~~~~~~~~~
test.rs:139:26: 139:38 error: the type of this value must be known in this context
test.rs:139                 assert!(!fp.is_null());
                                     ^~~~~~~~~~~~
test.rs:139:17: 139:40 note: in this expansion of assert! (defined in <std macros>)
test.rs:153:17: 153:29 error: the type of this value must be known in this context
test.rs:153             if !gv.is_null() {
                            ^~~~~~~~~~~~
test.rs:156:26: 156:38 error: the type of this value must be known in this context
test.rs:156                 assert!(!gp.is_null());
                                     ^~~~~~~~~~~~
test.rs:156:17: 156:40 note: in this expansion of assert! (defined in <std macros>)
error: aborting due to 7 previous errors
```

**Comment by alexcrichton** [comment]
r=me once this is passing, thanks @eddyb!

File: src/librustc_driver/lib.rs [hunk 1]
```diff
@@ -166,6 +166,10 @@ pub fn run_compiler<'a>(args: &[String],
 
     let sopts = config::build_session_options(&matches);
 
+    if sopts.debugging_opts.debug_llvm {
```
Comment (nagisa) [inline src/librustc_driver/lib.rs]: Could be moved all the way to phase 3 I think?

File: src/librustc_platform_intrinsics/Cargo.toml [hunk 2]
```diff
@@ -7,7 +7,3 @@ version = "0.0.0"
 name = "rustc_platform_intrinsics"
 path = "lib.rs"
 crate-type = ["dylib"]
-
-[dependencies]
```
Comment (nagisa) [inline src/librustc_platform_intrinsics/Cargo.toml]: Awesome!

File: src/librustc_driver/lib.rs (hunk 1 above)
Comment (eddyb) [inline src/librustc_driver/lib.rs]: Just in case something is touching LLVM before then, I didn't want to change behavior.

**Comment by eddyb** [comment]
@arielb1 Heh, I put this up to get some thoughts and didn't get to make sure everything compiles, thanks!

**Pull request merged**